app/Http/Controllers/BlogController.php

```diff
@@ -2,47 +2,278 @@
-    $post = $post->getPost($session, $id);
-    return view('blog.post', ['post' => $post]);
-  }
-
-  public function getPosts(Store $session, Request $request)
-  {
-    $post = new Post();
-    return $post;
-  }
-
-  public function postAdminCreate(Store $session, Request $request)
-  {
-    $this->validate($request, [
-        'title' => 'required|min:5',
-        'content' => 'required|min:10'
-    ]);
-
-    $post = new Post([
-      'title' => $request->input('title'),
-      'content' => $request->input('content')
-    ]);
-
-    $post->save();
-    return redirect()->route('blog.index');
-  }
+
+  public function __construct()
+    {
+        $this->middleware('jwt.auth', ['except' => 'getAllPosts']);
+    }
+  /**
+     * Get posts for a user
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function getAllUserPosts(Request $request)
+    {
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
+            return response()->json(['message' => 'User not found'], 404);
+        }
+
+        $user_id = $user->id;
+
+        $user = User::where("id", $user_id)->first();
+        $userPost = $user->blogPosts()->where('user_id', $user_id)->orderBy('created_at', 'desc')->get();
+        return $userPost;
+    }
+
+    /**
+     * Get a posts 
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function getOnePost(Request $request, $postId)
+    {
+        $post = Post::where("id", $postId)->first();   
+        return response()->json(["success" => true, "post" => $post]);
+    }
+
+    /**
+     * Create posts for a user
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function createPost(Request $request)
+    {
+        
+        $this->validate($request, [
+            'title' => 'required|min:5',
+            'content' => 'required|min:10'
+        ]);
+
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
+          return response()->json(['message' => 'User not found'], 404);
+        }
+
+        $user_id = $user->id;
+        $currentUser = User::where("id", $user_id)->first();
+    
+        $post = new Post([
+          'title' => $request->input('title'),
+          'content' => $request->input('content'),
+          "author" => $currentUser->username
+        ]);
+        
+        $currentUser->blogPosts()->save($post);
+
+        return response()->json(['success' => true, 'The blog post has been created']);
+    }
+
+    /**
+     * Edit a created blog post
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function editPost(Request $request, $postId)
+    {
+        
+        $this->validate($request, [
+            'title' => 'required|min:5',
+            'content' => 'required|min:10'
+        ]);
+
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
```

Comment:
I'm guessing there should be a better way to do this in Lumen. You can use `middleware` auth or something similar.

You can otherwise abstract this into a method that you can reuse instead of declaring it in every method. You can take advantage of constructors as well.